runtime/src/main/java/org/corfudb/runtime/object/transactions/Transaction.java

```diff
@@ -0,0 +1,75 @@
+package org.corfudb.runtime.object.transactions;
+
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Getter;
+import lombok.Setter;
+import org.corfudb.protocols.wireprotocol.Token;
+import org.corfudb.runtime.CorfuRuntime;
+
+
+/**
+ * Represents a transaction.
+ *
+ * <p>Created by Maithem on 12/5/18.
+ */
+
+@Getter
+@AllArgsConstructor
+public class Transaction {
+
+    /** The runtime for the context.
+     *
+     */
+    final CorfuRuntime runtime;
+
+    /** The type of context to build.
+     *
+     */
+    final TransactionType type;
+
+    /** For snapshot transactions, the address the
+     * snapshot will start at.
+     */
+    final Token snapshot;
+
+    /** Start the transaction with the parameters given
+     * to the builder.
+     */
+    public void begin() {
+        //TODO(Maithem) move timestamp validation here
+        TransactionalContext.newContext(type.get.apply(this));
+    }
+
+    public static TransactionBuilder builder() {
+        return new TransactionBuilder();
+    }
+
+    public static class TransactionBuilder {
```

Comment:
Please use lombok `@Builder`, this code is boilerplate

Reply:
Done. Thanks for the advice.